### PR TITLE
[poco] Update to 1.10.1

### PR DESCRIPTION
**Pull request by PhoebeHui**
Related to https://github.com/microsoft/vcpkg/issues/9964

1. Update poco to latest release revision 1.10.1
2. Update the patch static_pcre.patch and use-vcpkg-libharu.patch to adjust the source changes.
3. Remove unused patch find_pcre.patch and fix_foundation_link.patch 
4. Add vcpkg-cmake-wrapper.cmake for pcre
5. Add patch fix_dependency.patch to fix the dependency issue for sqlite3, pcre and expat.

Test:
find_package(Poco REQUIRED JWT Net XML Zip)
target_link_libraries(main PRIVATE Poco::JWT  Poco::Net Poco::XML Poco::Zip)

All features test pass with x64-windows. 
Skip features test on x64-linux since the dependency port is broken.

### CODE REVIEW

**Comment by KevDi** [comment]
Is there currently a plan when the Poco 1.10.1 Release is available in vcpkg?

**Comment by PhoebeHui** [comment]
I will continue to update this PR this week.

**Comment by BillyONeal** [comment]
@PhoebeHui Looks like it's not happy :/

**Comment by PhoebeHui** [comment]
The dependency port expat has usage issue which will be fixed by PR https://github.com/microsoft/vcpkg/pull/13022, I will continue this work after the PR merged.

**Comment by PhoebeHui** [comment]
The json5-parser failures would be fixed by https://github.com/microsoft/vcpkg/pull/10326

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

File: ports/pcre/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+include(FindPackageHandleStandardArgs)
```
Comment (BillyONeal) [inline ports/pcre/vcpkg-cmake-wrapper.cmake:1]: Quoth @ras0219 "vcpkg_cmake_wrapper" should only be used when we are overriding the behavior of a shipped-with-CMake find_library module, but "FindPCRE" is not a built in cmake module. We should patch the build system in question to `install(EXPORT` the bits someone wants instead.

Comment (PhoebeHui) [reply ports/pcre/vcpkg-cmake-wrapper.cmake:1]: Thanks for your review! I will update it.

**Comment by ras0219-msft** [comment]
Ping @PhoebeHui in case this was dropped

**Comment by PhoebeHui** [comment]
This PR depends on PR https://github.com/microsoft/vcpkg/pull/13539.

File: ports/pcre/vcpkg-cmake-wrapper.cmake (hunk 1 above)
Comment (PhoebeHui) [reply ports/pcre/vcpkg-cmake-wrapper.cmake]: Updated! PR #13539 has been merged, and I already merged the changes from master.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**